src/osd/OpRequest.cc

```diff
@@ -93,6 +93,14 @@ void OpRequest::_dump_op_descriptor_unlocked(ostream& stream) const
   get_req()->print(stream);
 }
 
+void OpRequest::_fetch_poolid(uint64_t& poolid)
+{
+  auto m = dynamic_cast<const MOSDFastDispatchOp*>(get_req());
+  if(m != nullptr) {
```

Comment:
Can it happen? i.e. - should it be an assert?

Reply:
I did encounter an issue of OSD asserting at the time of creating PGs last week but failed to debug it further. I have tried to reproduce this issue today but wasn't able to reproduce it.